src/test/scala/com/linkedin/feathr/offline/SlidingWindowAggIntegTest.scala

```diff
@@ -983,4 +983,87 @@ class SlidingWindowAggIntegTest extends FeathrIntegTest {
 
     validateRows(dfs.select(keyField, features: _*).collect().sortBy(row => row.getAs[Int](keyField)), expectedRows)
   }
+
+
+  @Test
+  def testSWACountDistinct(): Unit = {
+    val featureDefAsString =
+      """
+        |sources: {
+        |  swaSource: {
+        |    location: { path: "generation/daily/" }
+        |    isTimeSeries: true
+        |    timeWindowParameters: {
+        |      timestampColumn: "timestamp"
+        |      timestampColumnFormat: "yyyy-MM-dd"
+        |    }
+        |  }
+        |}
+        |anchors: {
+        |  swaAnchorWithKeyExtractor: {
+        |    source: "swaSource"
+        |    key: [x]
+        |    features: {
+        |      f: {
+        |        def: "Id"   // the column that contains the raw view count
+        |        aggregation: COUNT
+        |        window: 10d
+        |      }
+        |      g: {
+        |        def: "Id"   // the column that contains the raw view count
+        |        aggregation: COUNT_DISTINCT
+        |        window: 10d
+        |      }
+        |    }
+        |  }
+        |}
+    """.stripMargin
+
+    val features = Seq("f", "g")
+    val keyField = "x"
+    val featureJoinAsString =
+      s"""
+         | settings: {
+         |  joinTimeSettings: {
+         |   timestampColumn: {
+         |     def: timestamp
+         |     format: yyyy-MM-dd
+         |   }
+         |  }
+         |}
+         |features: [
+         |  {
+         |    key: [$keyField],
+         |    featureList: [${features.mkString(",")}]
+         |  }
+         |]
+    """.stripMargin
+
+
+    /**
+     * Expected output:
+     * +--------+----+----+
+     * |x|   f|   g|
+     * +--------+----+----+
+     * |       1|   6|   2|
+     * |       2|   6|   2|
+     * |       3|   1|   1|
```

Comment:
doesn't match the code(expectedRows)

Reply:
@hangfei good catch. I loaded the test data into python to double check the results are expected. Curious how your team validates the results.

Reply:
Are you asking about unit test?
@ahlag added a pretty nice dev guide for scala unit test: https://github.com/linkedin/feathr/blob/main/docs/dev_guide/scala_dev_guide.md. You can try it out.

We also have github actions to kick off e2e testing.

Reply:
Sweet. I will take a look!

Reply:
@esadler-hbo 
Please let me know if there’s any question or anything I left out.

Reply:
@ahlag super helpful doc! Thank you!

Reply:
@esadler-hbo fixed in the below commit (https://github.com/linkedin/feathr/pull/594/commits/1db050c7b9692158bd9ecdcf87ae9e41763900e0)

Reply:
resolving the conversation now